v0/destinations/mp/transform.js

```diff
@@ -158,107 +197,33 @@ function processTrack(message, destination) {
   return returnValue;
 }
 
-function getTransformedJSON(message, mappingJson, useNewMapping) {
-  let rawPayload = constructPayload(message, mappingJson);
-  if(isDefined(rawPayload.$geo_source) && (!GEO_SOURCE_ALLOWED_VALUES.includes(rawPayload.$geo_source))) {
-    throw new CustomError ("$geo_source value must be either null or 'reverse_geocoding' ", 400);
-  }
-  const userName = get(rawPayload, "$name");
-  if (!userName) {
-    set(rawPayload, "$name", getFullName(message));
-  }
+async function processIdentifyEvents(message, type, destination) {
+  let returnValue;
 
-  rawPayload = extractCustomFields(
+  // Creating the response to identify an user
+  // https://developer.mixpanel.com/reference/profile-set
+  returnValue = createIdentifyResponse(
     message,
-    rawPayload,
-    ["traits", "context.traits"],
-    MP_IDENTIFY_EXCLUSION_LIST
+    type,
+    destination,
+    responseBuilderSimple
   );
-  /*
-  we are adding backward compatibility using useNewMapping key.
-  TODO :: This portion need to be removed after we deciding to stop 
-  support for old mapping.
-  */
-
-  if (!useNewMapping) {
-    if (rawPayload.$first_name) {
-      rawPayload.$firstName = rawPayload.$first_name;
-      delete rawPayload.$first_name;
-    }
-    if (rawPayload.$last_name) {
-      rawPayload.$lastName = rawPayload.$last_name;
-      delete rawPayload.$last_name;
-    }
-  }
-
-  return rawPayload;
-}
 
-function processIdentifyEvents(message, type, destination) {
-  const returnValue = [];
-  // this variable is used for supporting backward compatibility
-  const { useNewMapping } = destination.Config;
-  // user payload created
-  let properties = getTransformedJSON(
-    message,
-    mPIdentifyConfigJson,
-    useNewMapping
-  );
-  const device = get(message, "context.device");
-  if (device && device.token) {
-    let payload;
-    if (isAppleFamily(device.type)) {
-      payload = constructPayload(message, mPProfileIosConfigJson);
-      properties.$ios_devices = [device.token];
-    } else if (device.type.toLowerCase() === "android") {
-      payload = constructPayload(message, mPProfileAndroidConfigJson);
-      properties.$android_devices = [device.token];
+  // If userId and anonymousId both are present and required credentials for /import
+  // endpoint are available we are creating the merging response below
+  // https://developer.mixpanel.com/reference/identity-merge
+  if (
+    message.userId &&
+    message.anonymousId &&
+    (destination.Config.apiSecret ||
+      (destination.Config.serviceAccountSecret &&
+        destination.Config.serviceAccountUserName &&
+        destination.Config.projectId))
```

Comment:
The config credentials related presence validation can be move to a mico function with proper name for ease of understanding

Reply:
Updated